services/messaging/src/main/java/org/eclipse/hono/messaging/ForwardingDownstreamAdapter.java

```diff
@@ -278,6 +275,13 @@ private void onDisconnectFromDownstreamContainer(final ProtonConnection con) {
         }
     }
 
+    private void closeReceiverOnDownstreamClose(final UpstreamReceiver receiver) {
```

Comment:
maybe better *closeReceiverOnDownstreamDetach*?

Reply:
Yes, the generic detach is more accurate.